src/pages/SourcePage.vue

```diff
@@ -130,15 +151,38 @@ const SourcePage = {
     HeaderSection,
     FooterSection,
   },
+  data() {
+    return {
+      sort: {
+        direction: 'asc',
+        field: 'display_name',
+      },
+    }
+  },
   computed: {
     imageProviders() {
       return this.$store.state.imageProviders
     },
+    sortedProviders() {
+      const sorted = sortBy(this.imageProviders, [this.sort.field])
+      return this.sort.direction === 'asc' ? sorted : sorted.reverse()
+    },
   },
   methods: {
     getProviderImageCount(imageCount) {
       return imageCount.toLocaleString('en')
     },
+    sortTable(field) {
+      let direction = 'asc'
+      if (field === this.sort.field) {
+        direction = this.sort.direction === 'asc' ? 'desc' : 'asc'
+      }
```

Comment:
here i only invert the sort direction if you're on the same column, otherwise 'reset' to the default, in this case 'asc'.